### PR TITLE
Moved and revised the value types article

### LINKED ISSUE #15997: Structs and shallow copy
Structs are copied on assignment. When a struct is assigned to a new variable, all the data is copied, and any modification to the new copy does not change the data for the original copy. This is important to remember when working with collections of value types such as Dictionary&lt;string, myStruct&gt;.

I found this line confusing since it does not mention anything about shallow copy and when we use List or Dictionary inside structs they will indeed be shallow copied. Maybe rephrasing the above mentioned paragraph would be better. Thanks

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: b58203aa-f1ff-0741-72a7-dd865a69b08a
* Version Independent ID: 35ebe8a9-4465-280e-082c-b3f74d031ac0
* Content: [Structs - C# Programming Guide](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/structs#feedback)
* Content Source: [docs/csharp/programming-guide/classes-and-structs/structs.md](https://github.com/dotnet/docs/blob/master/docs/csharp/programming-guide/classes-and-structs/structs.md)
* Product: **dotnet-csharp**
* Technology: **csharp-fundamentals**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

### PR DESCRIPTION
**Pull request by pkulikov**
Moved and revised the [Value types](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/value-types) article. More focus on the copy semantics, with examples. Removed the section about initializing because that information is covered by examples in the articles about individual types.

Redirected the [Value types table](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/value-types-table) to the new article.

Fixes #15997 (I'm moving that discussion from the article about structure types)
Contributes to #10726 
Supported by dotnet/samples#1923

### CODE REVIEW

File: docs/csharp/language-reference/builtin-types/value-types.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+## Kinds of value types
+
+A value type can be one of the two following kinds:
+
+- a [structure type](../keywords/struct.md), which encapsulates data and related functionality
+- an [enumeration type](enum.md), which is defined by a set of named constants and represents a choice or a combination of choices
+
+A [nullable value type](nullable-value-types.md) `T?` represents all values of its underlying value type `T` and an additional [null](../keywords/null.md) value. You cannot assign `null` to a variable of a value type, unless it's a nullable value type.
+
+## Built-in value types. Simple types
```
Comment (Youssef1313) [inline docs/csharp/language-reference/builtin-types/value-types.md]: Looks better to me with parenthesis.

```suggestion
## Built-in value types (simple types)
```

File: docs/csharp/language-reference/builtin-types/value-types.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  - "value types [C#]"
+  - "types [C#], value types"
+  - "C# language, value types"
+ms.assetid: 471eb994-2958-49d5-a6be-19b4313f80a3
+---
+# Value types (C# reference)
+
+*Value types* and [reference types](../keywords/reference-types.md) are the two main categories of C# types. A variable of a value type contains an instance of the type. This differs from a variable of a reference type, which contains a reference to an instance of the type. By default, on [assignment](../operators/assignment-operator.md), passing an argument to a method, or returning a method result, variable values are copied. In the case of value-type variables, the corresponding type instances are copied. The following example demonstrates that behavior:
+
+[!code-csharp[copy of values](~/samples/csharp/language-reference/builtin-types/ValueTypes.cs#ValueTypeCopied)]
```
Comment (Youssef1313) [inline docs/csharp/language-reference/builtin-types/value-types.md]: Should the new code reference syntax be used ?

https://docs.microsoft.com/en-us/contribute/how-to-write-use-markdown#snippet-syntax-reference

```suggestion
:::code language="csharp" source="~/samples/csharp/language-reference/builtin-types/ValueTypes.cs" id="ValueTypeCopied":::
```

File: docs/csharp/language-reference/builtin-types/value-types.md (hunk 1 above)
Comment (Youssef1313) [reply docs/csharp/language-reference/builtin-types/value-types.md]: Got you. Then, what do you think about having simple types as h3 under the Built-in value types ?
That way, it looks more clearer

Comment (pkulikov) [reply docs/csharp/language-reference/builtin-types/value-types.md]: That would work. But now I think if the docs even should introduce and describe simple types. The link to spec would work if necessary. @BillWagner what do you think?

File: docs/csharp/language-reference/builtin-types/value-types.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+As the preceding example shows, operations on a value-type variable don't affect other variables.
```
Comment (BillWagner) [inline docs/csharp/language-reference/builtin-types/value-types.md]: I'd like to rephrase this so it highlights the distinction between *variables* and *the thing a variable refers to*:

```suggestion
As the preceding example shows, operations on a value-type variable affect only that instance of the value type, stored in the variable.
```

File: docs/csharp/language-reference/builtin-types/value-types.md (hunk 1 above)
Comment (BillWagner) [reply docs/csharp/language-reference/builtin-types/value-types.md]: I think the heading should be just "Built in". This section covers both, and the distinction isn't needed for the heading:

```suggestion
## Built-in value types
```

**Review by BillWagner — CHANGES_REQUESTED** [review]
This looks great @pkulikov 

I think there are a couple small updates to make, and then this is ready.

**Comment by pkulikov** [comment]
@BillWagner @Youssef1313 thanks for the review; I've applied all the suggestions.

**Review by BillWagner — APPROVED** [review]

**Comment by BillWagner** [comment]
Thanks @pkulikov 

I'll :shipit: now.

**Pull request merged**